WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductSelectorView.swift

```diff
@@ -157,10 +186,17 @@ struct ProductSelectorView: View {
             ToolbarItem(placement: .cancellationAction) {
                 if let cancelButtonTitle = configuration.cancelButtonTitle {
                     Button(cancelButtonTitle) {
-                        isPresented.toggle()
-                        if !isPresented {
+                        // When using navigation controller, the dismissal is handled by the parent navigation controller,
+                        // so we call the callback immediately.
+                        if usingNavigationController {
```

Comment:
This feels a little hack-ish to me but I can't figure out the better way to do it.

The crux of the issue is that this Button by default toggles `isPresented` and checks for it before calling the callback, but when using navigation controller, we do not care about `isPresented` state.

Reply:
I believe checking `isPresented` before dismissing is redundant. You can remove `usingNavigationController` and keep the logic simple for all cases:
```Swift
isPresented = false
viewModel.closeButtonTapped()
```

Reply:
Thanks, you're right. I was avoiding to modify old code but this change works fine in my check.

https://github.com/woocommerce/woocommerce-ios/pull/11625/commits/abc91c23c2506fb7d0a75446fd7725b6a5c82b03